src/NNCache.cpp

```diff
@@ -0,0 +1,88 @@
+/*
+    This file is part of Leela Zero.
+    Copyright (C) 2017 Michael O
+
+    Leela Zero is free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    Leela Zero is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with Leela Zero.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include "NNCache.h"
+
+#include <functional>
+
+#include "Utils.h"
+
+NNCache::NNCache(int size) : m_size(size) {}
+
+NNCache* NNCache::get_NNCache(void) {
+    static NNCache* cache = new NNCache();
```

Comment:
Prefer a reference to a static here. (magic static)

Reply:
fixed